src/components/Documentation/RightPanel/index.tsx

```diff
@@ -34,6 +35,31 @@ const RightPanel: React.FC<IRightPanelProps> = ({
   const [currentHeadingSlug, setCurrentHeadingSlug] = useState<string | null>(
     null
   )
+  const generateHeadingId = (slug: string): string => `links-${slug}`
+  const tableOfContentsRefs: {
+    [key: string]: React.RefObject<HTMLDivElement>
+  } = headings
+    .map(({ slug }) => ({
+      id: generateHeadingId(slug),
+      ref: createRef()
+    }))
+    .reduce((acc, curr) => {
+      return {
+        ...acc,
+        [curr.id]: curr.ref
+      }
+    }, {})
```

Comment:
This dynamically creates React ref for us to get references to the link DOM elements.

Reply:
Personally I'd do what the `map` does in a single `reduce`, but that's just a stylistic nitpick with a likely negligible performance hit.

Reply:
I agree, that is cleaner. Updated in https://github.com/kangzeroo/dvc.org/pull/1/commits/a44334d073dbc212ad1d73ad58daba10537de0dc